### ISSUE TITLE
gclient sync fails on mac

### ISSUE THREAD

**Issue by ianloic** [domokit/mojo#802]
Rust doesn't work on Mac or something.

```
Syncing projects: 100% (35/35), done.

________ running '/usr/bin/python src/build/landmines.py' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/tools/clang/scripts/update.py --if-needed' in '/Users/ianloic/tq/mojo'
Clang already at 259396-1

________ running '/usr/bin/python src/tools/dart/update.py' in '/Users/ianloic/tq/mojo'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 21.2M  100 21.2M    0     0  5770k      0  0:00:03  0:00:03 --:--:-- 7865k
Running ['curl', '-C', '-', '--location', '-o', '/Users/ianloic/tq/mojo/src/third_party/dart-sdk/dartsdk-macos-x64-release.zip', 'https://gsdview.appspot.com/dart-archive/channels/dev/raw/1.18.0-dev.2.0/sdk/dartsdk-macos-x64-release.zip']
Running ['unzip', '-o', '-q', '/Users/ianloic/tq/mojo/src/third_party/dart-sdk/dartsdk-macos-x64-release.zip', '-d', '/Users/ianloic/tq/mojo/src/third_party/dart-sdk']

________ running '/usr/bin/python src/tools/android/download_android_tools.py' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/build/download_sdk_extras.py' in '/Users/ianloic/tq/mojo'

________ running 'src/tools/download_from_google_storage.py --no_resume --quiet --platform=linux* --no_auth --bucket chromium-gn -s src/buildtools/linux64/gn.sha1' in '/Users/ianloic/tq/mojo'

________ running 'src/tools/download_from_google_storage.py --no_resume --quiet --platform=darwin --no_auth --bucket chromium-gn -s src/buildtools/mac/gn.sha1' in '/Users/ianloic/tq/mojo'
0> Downloading src/buildtools/mac/gn...

________ running 'src/tools/download_from_google_storage.py --no_resume --quiet --platform=win* --no_auth --bucket chromium-gn -s src/buildtools/win/gn.exe.sha1' in '/Users/ianloic/tq/mojo'

________ running 'src/tools/download_from_google_storage.py --no_resume --quiet --platform=linux* --no_auth --bucket chromium-clang-format -s src/buildtools/linux64/clang-format.sha1' in '/Users/ianloic/tq/mojo'

________ running 'src/tools/download_from_google_storage.py --no_resume --quiet --platform=darwin --no_auth --bucket chromium-clang-format -s src/buildtools/mac/clang-format.sha1' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/third_party/binutils/download.py' in '/Users/ianloic/tq/mojo'

________ running 'src/tools/download_from_google_storage.py --no_resume --quiet --platform=linux* --no_auth --bucket chromium-eu-strip -s src/build/linux/bin/eu-strip.sha1' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/mojo/public/tools/download_network_service.py --tools-directory ../../../tools' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/tools/remove_stale_pyc_files.py src/tools' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/build/download_nacl_toolchains.py --packages pnacl_newlib,pnacl_translator sync --extract' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/tools/go/download.py' in '/Users/ianloic/tq/mojo'

________ running '/usr/bin/python src/tools/rust/download.py' in '/Users/ianloic/tq/mojo'
Traceback (most recent call last):
  File "src/tools/rust/download.py", line 102, in <module>
    sys.exit(main())
  File "src/tools/rust/download.py", line 91, in main
    with open(os.path.join(THIS_DIR, VersionFileName())) as f:
  File "src/tools/rust/download.py", line 43, in VersionFileName
    raise Exception('unsupported platform: ' + sys.platform)
Exception: unsupported platform: darwin
Error: Command '/usr/bin/python src/tools/rust/download.py' returned non-zero exit status 1 in /Users/ianloic/tq/mojo
```

**Comment by mikejurka** [comment]
disclaimer: i'm not on the mojo team, i'm using it as a member of the sysui team

Have you been able to compile mojo on mac at all? That hasn't been working for me for months.

If all you care about is syncing (but not compiling), I've used this as a workaround:

Add a "return 0" on at
tools/rust/download.py:89 

so it looks like:

```
def main():
  # Read latest version.
  return 0
```

**Comment by mknyszek** [comment]
Sorry about that! I'll submit a fix for that ASAP.

The workaround seems fine and it's pretty much what I'll end up doing for now. I could get the Rust bindings to support Mac fairly easily (all I need is the Mac toolchain hosted), but if Mojo on Mac isn't building at all then there doesn't really seem to me to be a point right now.

If anyone has any thoughts on whether I should try anyway though, let me know!

**Comment by mikejurka** [comment]
Please try to get it running on mac, b/c otherwise the future task of getting compiles working on mac will just be that much more complicated.

**Comment by gabeschine** [comment]
There are a lot of us that use Mac as our main dev environment.

**Comment by mikejurka** [comment]
@gabeschine not totally related to this issue, but are you building mojo on Mac? (I'm forced to build on Linux but still use Mac for editing)

**Comment by ianloic** [comment]
Mojo doesn't build on the mac but it should at least check out.

**Comment by gabeschine** [comment]
Well, I'm actually building *Modular* on mac, which depends on Mojo, but I
believe our requirements are somewhat different. I did at one point do Mojo
development on Mac but that was >6 months ago.

**Comment by mikejurka** [comment]
@mknyszek  Seems like quite a few of us rely on a mojo checkout on Mac. Please commit a workaround, or even better, get the rust build running on Mac. Until then, we'll have to use the workaround.

**Comment by mknyszek** [comment]
Working on just having the toolchain pull in for Mac. I'll flip the build switches too and make sure it keeps building with my future changes. I should have a patch out for review today, and it should hopefully land early next week.

**Comment by mknyszek** [comment]
I just landed a workaround that I tested on Mac. The bindings won't build out of the box yet since I didn't flip any build switches (they'll just be ignored at build time). However, manually trying them out results in them building successfully! There are some oddities with Cargo on Mac it seems, so I'll defer hooking it into the build system until a little later, after I upgrade to Rust 1.10.

Let me know if you run into any issues! I'll close the issue for now.
